studio/pages/integrations/github/install.tsx

```diff
@@ -0,0 +1,198 @@
+import { useRouter } from 'next/router'
+import { useMemo, useState } from 'react'
+
+import { useParams } from 'common'
+import OrganizationPicker from 'components/interfaces/Integrations/OrganizationPicker'
+import { Markdown } from 'components/interfaces/Markdown'
+import GitHubIntegrationWindowLayout from 'components/layouts/IntegrationsLayout/GitHubIntegrationWindowLayout'
+import { getHasInstalledObject } from 'components/layouts/IntegrationsLayout/Integrations.utils'
+import { ScaffoldColumn, ScaffoldContainer, ScaffoldDivider } from 'components/layouts/Scaffold'
+import { useGitHubIntegrationCreateMutation } from 'data/integrations/github-integration-create-mutation'
+import { useIntegrationsQuery } from 'data/integrations/integrations-query'
+import { useOrganizationsQuery } from 'data/organizations/organizations-query'
+import { useStore } from 'hooks'
+import { NextPageWithLayout, Organization } from 'types'
+import {
+  Alert,
+  AlertDescription_Shadcn_,
+  AlertTitle_Shadcn_,
+  Alert_Shadcn_,
+  Button,
+  IconAlertTriangle,
+  IconBook,
+  IconLifeBuoy,
+  LoadingLine,
+} from 'ui'
+
+const GitHubIntegration: NextPageWithLayout = () => {
+  const router = useRouter()
+  const { ui } = useStore()
+  const { installation_id: installationId } = useParams()
+  const [selectedOrg, setSelectedOrg] = useState<Organization | null>(null)
+  const [organizationIntegrationId, setOrganizationIntegrationId] = useState<string | null>(null)
+
+  /**
+   * Fetch the list of organization based integration installations for GitHub.
+   *
+   * Array of integrations installed on all
+   */
+  const { data: integrationData } = useIntegrationsQuery()
+
+  const { data: organizationsData } = useOrganizationsQuery({
+    onSuccess(organizations) {
+      const firstOrg = organizations?.[0]
+      if (firstOrg && selectedOrg === null) {
+        setSelectedOrg(firstOrg)
+        router.query.organizationSlug = firstOrg.slug
+      }
+    },
+  })
+
+  const { mutate, isLoading: isLoadingGitHubIntegrationCreateMutation } =
+    useGitHubIntegrationCreateMutation({
+      onSuccess({ id }) {
+        setOrganizationIntegrationId(id)
+
+        router.push({
+          pathname: `/integrations/github/${id}/choose-project`,
+          query: router.query,
+        })
+      },
+      onError(error: any) {
```

Comment:
Shift error handler into the mutation instead in `github-integration-create-mutation`

Reply:
Not quite sure what you mean sorry. There's a default error handler there now and this will override that?

Reply:
oh wait sorry i must've seen wrongly 🤦 yeah github-integratio-create-mutation has a default error handler so i think we can remove this since its doing the same thing

but we'll need to update the toast message of the error handler within the mutation to what you've got here "Failed to create Github integration", rather than "Failed to mutate"